pkg/reconciler/isbsvc/controller.go

```diff
@@ -122,13 +126,14 @@ func (r *interStepBufferServiceReconciler) reconcile(ctx context.Context, isbSvc
 	return installer.Install(ctx, isbSvc, r.client, r.kubeClient, r.config, log, r.recorder)
 }
 
-func (r *interStepBufferServiceReconciler) needsUpdate(old, new *dfv1.InterStepBufferService) bool {
-	if old == nil {
-		return true
+func needsToPatchFinalizers(old, new *dfv1.InterStepBufferService) bool {
+	if old == nil { // This is a weird scenario, nothing we can do. Theoretically it will never happen.
```

Comment:
since it is a weird scenario should be log so it might help whoever is debugging?

Reply:
Cleaned it up since there's no need to add this function.